patch/patcher.go

```diff
@@ -276,14 +280,24 @@ func (p *Patcher) scanField(f *protogen.Field) {
 		}
 		newName = lint.Name(newName, lints.InitialismsMap())
 	}
+	// check embed for message types
+	embed := f.Message != nil && f.Oneof == nil && opts.GetEmbed()
```

Comment:
This still needs a warning log if user attempts to embed a non-message or a oneof field.